cmd/genji/shell/command.go

```diff
@@ -2,20 +2,94 @@ package shell
 
 import (
 	"fmt"
+	"os"
 	"sort"
+	"strings"
 
+	"github.com/agnivade/levenshtein"
 	"github.com/genjidb/genji"
 	"github.com/genjidb/genji/document"
 )
 
-var commands = map[string]string{
-	".tables":               "\t\tList names of tables.",
-	".exit":                 "\t\t\tExit this program.",
-	".indexes [table_name]": "\tDisplay all indexes or the indexes of the given table name.",
-	".help":                 "\t\t\tList all commands.",
+type Command struct {
+	Name    string
+	Options string
+	Help    string
+	Action  func(db *genji.DB, cmd []string) error
 }
 
-func runTablesCmd(db *genji.DB, cmd []string) error {
+type lessFunc func(a, b *Command) int
```

Comment:
I don't think we need to go that far for dealing with sorting. There are barely 4 commands we can definitely chose the order manually. Also, I don't think alphabetical order is the best way to sort these commands, I think it would be better to group them by categories in the future.

Reply:
I am agree and I like the "groupby" categories.